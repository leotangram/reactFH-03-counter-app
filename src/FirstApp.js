import React from 'react'

const FirstApp = () => {
  const greeting = 'Hola mundo'

  // const person = {
  //   name: 'Leo',
  //   age: 34
  // }

  return (
    <>
      <h1>{greeting}</h1>
      {/* <pre>{JSON.stringify(person, null, 3)}</pre> */}
      <p>Mi primera aplicaci√≥n</p>
    </>
  )
}

export default FirstApp
