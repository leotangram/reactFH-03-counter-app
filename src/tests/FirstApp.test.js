import React from 'react'
// import { render } from '@testing-library/react'
import '@testing-library/jest-dom'
import { shallow } from 'enzyme'
import FirstApp from '../FirstApp'

describe('<FirstApp />', () => {
  // test('should show the message "Hola, soy Goku"', () => {
  //   const greeting = 'Hola, soy Goku'
  //   const { getByText } = render(<FirstApp greeting={greeting} />)
  //   expect(getByText(greeting)).toBeInTheDocument()
  // })

  test('should show <FirstApp /> correctly', () => {
    const greeting = 'Hola, soy Goku'
    const wrapper = shallow(<FirstApp greeting={greeting} />)
    expect(wrapper).toMatchSnapshot()
  })

  test('should show the subtitle sent by props', () => {
    const greeting = 'Hola, soy Goku'
    const subtitle = 'Subt√≠tulo'
    const wrapper = shallow(
      <FirstApp greeting={greeting} subtitle={subtitle} />
    )
    const textParagraph = wrapper.find('p').text()
    expect(textParagraph).toBe(subtitle)
  })
})
