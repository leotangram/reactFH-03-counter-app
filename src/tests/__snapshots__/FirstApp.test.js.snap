// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<FirstApp /> should show <FirstApp /> correctly 1`] = `
<Fragment>
  <h1>
    Hola, soy Goku
  </h1>
  <p>
    Soy un subtitulo
  </p>
</Fragment>
`;

exports[`FirstApp should show <FirstApp /> correctly 1`] = `
<Fragment>
  <h1>
    Hola, soy Goku
  </h1>
  <p>
    Soy un subtitulo
  </p>
</Fragment>
`;
